lib/spack/spack/util/module_cmd.py

```diff
@@ -24,6 +23,17 @@
 def module(*args, **kwargs):
     module_cmd = kwargs.get("module_template", "module " + " ".join(args))
 
+    # On ubuntu the default shell `/bin/sh`` is `dash`. As `bin/spack` uses `/bin/sh` there are no
+    # bash functions and hence no `module` command. We replace the `module` command by the actual
+    # function call in the bash function.
+    if not any(x.startswith("BASH_FUNC_module") for x in os.environ):
+        if os.environ["MODULES_CMD"]:
```

Comment:
`os.environ.get`

Reply:
In ubuntu I see 
```
BASH_FUNC_module%%=() {  _module_raw "$@" 2>&1
}
```
and on AL2 I see
```
BASH_FUNC_module()=() {  eval `/usr/bin/modulecmd bash $*`
}
```
I took that syntax from here: https://github.com/stephenmsachs/spack/blob/ubuntu-dash/lib/spack/spack/test/environment_modifications.py#L512

Reply:
Or do you mean line 30?

Reply:
I mean that that branch on line 30 is incorrect, you never hit the else branch, it raises KeyError instead.

Reply:
got it. Thanks.